### PR TITLE
Feedback resource phase2

### PR DESCRIPTION
**Pull request by RonAzar**
•Implemented API endpoints for get Status, update Status, get Filtered Feedbacks

### CODE REVIEW

File: src/main/kotlin/com/hibob/academy/employee_feedback_feature/resource/FeedbackResource.kt [hunk 1]
```diff
@@ -53,7 +58,58 @@ class FeedbackResource(private val feedbackService: FeedbackService) {
         return if (hasPermission(employeeRole, Permissions.VIEW_ALL_FEEDBACKS)) {
             Response.ok(feedbackService.getAllFeedbacks(companyId)).build()
         } else {
-            Response.status(Response.Status.FORBIDDEN).entity("UNAUTHORIZED: You do not have access to view feedbacks").build()
+            Response.status(Response.Status.UNAUTHORIZED)
```
Comment (vasofi) [inline src/main/kotlin/com/hibob/academy/employee_feedback_feature/resource/FeedbackResource.kt:61]: same comment about the if permission (like in the other pr).

you can do - if (no permission) return
rest of the code not in if block

Comment (RonAzar) [reply src/main/kotlin/com/hibob/academy/employee_feedback_feature/resource/FeedbackResource.kt:61]: Fixed!

File: src/main/kotlin/com/hibob/academy/employee_feedback_feature/resource/FeedbackResource.kt [hunk 2]
```diff
@@ -53,7 +58,60 @@ class FeedbackResource(private val feedbackService: FeedbackService) {
...
     }
+
+    @Path("view/filter")
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    fun getFilteredFeedbacks(@Context requestContext: ContainerRequestContext, filter: FeedbackFilter): Response {
+        val companyId = extractClaimAsLong(requestContext, "companyId")!!
+        val employeeRole = extractRole(requestContext)!!
+
+        return if (hasPermission(employeeRole, Permissions.VIEW_ALL_FEEDBACKS)) {
```
Comment (vasofi) [inline src/main/kotlin/com/hibob/academy/employee_feedback_feature/resource/FeedbackResource.kt:74]: return with if - it's not clear.

also - first do the permission validation and return not authorised.
Then you can continue to the logic itself, without an if.

Or, another option, hasPermissions can throw the unauthorised exception, and then you don't need this if

File: src/main/kotlin/com/hibob/academy/employee_feedback_feature/resource/FeedbackResource.kt [hunk 3]
```diff
@@ -53,7 +58,60 @@ class FeedbackResource(private val feedbackService: FeedbackService) {
...
+    @Produces(MediaType.APPLICATION_JSON)
+    fun getStatus(
+        @Context requestContext: ContainerRequestContext,
+        @PathParam("feedbackId") feedbackId: Long
+    ): Response {
+        val role = extractRole(requestContext)!!
+        val companyId = extractClaimAsLong(requestContext, "companyId")!!
+        val searchedFeedback = SearchedFeedback(companyId, feedbackId)
+
+        if (!hasPermission(role, Permissions.VIEW_ALL_FEEDBACKS)) {
```
Comment (vasofi) [inline src/main/kotlin/com/hibob/academy/employee_feedback_feature/resource/FeedbackResource.kt:93]: you might get a conflict here due to the change in pr 51

File: src/main/kotlin/com/hibob/academy/employee_feedback_feature/resource/FeedbackResource.kt [hunk 4]
```diff
@@ -53,7 +58,60 @@ class FeedbackResource(private val feedbackService: FeedbackService) {
...
+
```
Comment (vasofi) [inline src/main/kotlin/com/hibob/academy/employee_feedback_feature/resource/FeedbackResource.kt:94]: no new line at the beginning of block, even if the beginning is a return

**Review by vasofi — APPROVED** [review]
fix the new lines
make sure you don't have conflicts

**Pull request merged**